src/main/java/org/alliancegenome/curation_api/services/AffectedGenomicModelService.java

```diff
@@ -197,14 +199,22 @@ private void handleSecondaryIds(AffectedGenomicModelDTO agm, AffectedGenomicMode
     }
     
     private boolean validateAffectedGenomicModelDTO(AffectedGenomicModelDTO agm) {
-        // TODO: add validation of taxon ID once taxons are loaded
+        // TODO: replace regex method with DB lookup for taxon ID once taxons are loaded
         
         // Check for required fields
         if (agm.getPrimaryID() == null || agm.getName() == null || agm.getTaxonId() == null) {
             log.debug("Entry for AGM " + agm.getPrimaryID() + " missing required fields - skipping");
             return false;
         }
         
+        // Validate taxon ID
+        Pattern taxonIdPattern = Pattern.compile("^NCBITaxon:\\d+$");
+        Matcher taxonIdMatcher = taxonIdPattern.matcher(agm.getTaxonId());
+        if (!taxonIdMatcher.find()) {
+            log.debug("Invalid taxon ID for AGM " + agm.getPrimaryID() + " - skipping");
+            return false;
+        }
+        
```

Comment:
A better way to do this might be to do a looking of the taxon in a HashSet<String> we are going to need taxons at some point... it will also get rid of the list thats in the Job Executer also.

Reply:
Yeah, this is only meant to be in the interim until we have taxons loaded.

Reply:
Ah, I see what you mean - I was forgetting that the taxons here are limited to the MOD taxons, unlike those in the experimental conditions.

Reply:
Either way... its fine... we need to get the taxons in the DB at some point.